// Copyright 2015 gRPC authors.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

//option java_multiple_files = true;
//option java_package = "io.common.ota.client";
//option java_outer_classname = "OtaClientProto";
option objc_class_prefix = "OTA";

package OtaClientV2;

// The OTA Client service definition.
// Style Guide: https://developers.google.com/protocol-buffers/docs/style#message_and_field_names
service OtaClientService {
  rpc Update (UpdateRequest) returns (UpdateResponse) {}
  rpc Rollback (RollbackRequest) returns (RollbackResponse) {}
  rpc Status (StatusRequest) returns (StatusResponse) {}
}

//
// Update
//

// Request
message UpdateRequestEcu {
  string ecu_id = 1;
  string version = 2;
  string url = 3;
  string cookies = 4;
}

message UpdateRequest {
  repeated UpdateRequestEcu update_request_ecu = 1;
}
// Common
enum Result {
  OK = 0;
  ERROR_OTA_STATUS = 1;
  ERROR_UNKNOWN = 255;
}

// Response
message UpdateResponseEcu {
  string ecu_id = 1;
  Result result = 2;
}

message UpdateResponse {
  repeated UpdateResponseEcu update_response_ecu = 1;
}

//
// Rollback
//

// Request
message RollbackRequestEcu {
  string ecu_id = 1;
  string version = 2;
  string url = 3;
  string cookies = 4;
}

message RollbackRequest {
  repeated RollbackRequestEcu rollback_request_ecu = 1;
}

// Response
message RollbackResponseEcu {
  string ecu_id = 1;
  Result result = 2;
}

message RollbackResponse {
  repeated RollbackResponseEcu rollback_response_ecu = 1;
}

//
// Status
//

// Request
message StatusRequestEcu {
  string ecu_id = 1;
  string version = 2;
  string url = 3;
  string cookies = 4;
}

message StatusRequest {
  repeated StatusRequestEcu status_request_ecu = 1;
}

// Response
message StatusResponseEcu {
  string ecu_id = 1;
  Result result = 2;
}

message StatusResponse {
  repeated StatusResponseEcu status_response_ecu = 1;
}


